### PR TITLE
Add patch to install missing header in APEX

### PR DESCRIPTION
**Pull request by msimberg**
This makes sure the `dependency_tree.hpp` header is installed (required from 2.3.2 onwards).

### CODE REVIEW

File: var/spack/repos/builtin/packages/apex/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/apex/package.py:74]: Can you turn this comment into a brief description of what the patch does and why it is needed?

**Review by alalazo — CHANGES_REQUESTED** [review]
Just a minor request

File: var/spack/repos/builtin/packages/apex/package.py
Comment (msimberg) [reply var/spack/repos/builtin/packages/apex/package.py:74]: I left the `# Patches` heading to fit the other sections (`# Dependencies`, `# Conflicts`, etc.), but added a comment specifically for this patch.

**Review by alalazo — DISMISSED** [review]

**Comment by khuck** [comment]
@msimberg Is this for an HPX build?  Because the header does get installed for normal builds.  I just checked CMakeLists.hpx, and the header is not installed during HPX builds.  It should be fixed there, but I suppose this patch would fix older versions...

**Comment by msimberg** [comment]
> @msimberg Is this for an HPX build? Because the header does get installed for normal builds. I just checked CMakeLists.hpx, and the header is not installed during HPX builds. It should be fixed there, but I suppose this patch would fix older versions...

This was for the standalone build, but I see it's been added now on develop. Thanks!

@khuck I can add an upper limit to when the patch is applied as well, which I guess would be 2.4.1 (the current release), assuming the next one is going to be 2.4.2?

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by alalazo** [comment]
@msimberg Let me know if you want to add an upper bound before merging

**Comment by msimberg** [comment]
> @msimberg Let me know if you want to add an upper bound before merging

I think it would be a good idea, but I'd like to wait for @khuck's response first.

**Comment by khuck** [comment]
Yeah, probably would be better to put an upper bound on it.  There will be a 2.4.2 release soon.  There should probably be a lower bound too - because that's a relatively new file.  2.3.2 was the first release where it appeared.

**Comment by msimberg** [comment]
> Yeah, probably would be better to put an upper bound on it. There will be a 2.4.2 release soon. There should probably be a lower bound too - because that's a relatively new file. 2.3.2 was the first release where it appeared.

Sounds good! The lower bound was already there. I've now added the upper bound as well. @alalazo do I remember correctly that `@2.3.2:2.4.1` is inclusive of both ends?

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
> do I remember correctly that @2.3.2:2.4.1 is inclusive of both ends?

Yes, that's correct. Thanks everyone!

**Pull request merged**
